packages/generator/ni_measurement_plugin_sdk_generator/client/templates/measurement_plugin_client.py.mako

```diff
@@ -72,7 +77,15 @@ class ${class_name}:
         if grpc_channel is not None:
             self._stub = v2_measurement_service_pb2_grpc.MeasurementServiceStub(grpc_channel)
         self._create_file_descriptor()
-        
+        self.pin_map_id: str = ""
+        self.sites: list[int] = [0]
```

Comment:
Here, sites is exposed a public property. Can we proceed with this? 
May I know your suggestions. @dixonjoel

Reply:
Is there any reason to expose it at this time? I would lean towards keeping it private.

Reply:
Made the `sites` as a read-only property by returning it as a `tuple` and added a setter to make it changeable by the users.

Reply:
> ... return a tuple of the sites and provide a "set" method to change it.

@mshafer-NI, any specific reasons to prefer a set method but not a property setter (@sites.setter) here?

Reply:
A property setter is implemented as a method -> and that just boils down to api -> does it make sense to be able to do "foo.sites = bar" or not.

Reply:
(And I did not mean to imply a decision for either `foo.set_sites(...)` or `foo.sites = ` via property setter -> this style choice should be driven by what "makes sense" in the context of what is getting set, and the rest of the API uses.)

Reply:
Resolving this thread. Please refer this [thread](https://github.com/ni/measurement-plugin-python/pull/871#discussion_r1752287928) for details.